packages/dds/tree/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -54,110 +54,48 @@ function rebaseMarkList<TNodeChange>(
-                        const [baseMarkBefore, baseMarkAfter] = splitMarkOnOutput(baseMark, offset);
-                        baseIter.push(baseMarkAfter);
-                        factory.pushOffset(getOutputLength(baseMarkBefore));
-                    }
-
-                    if (offset < reattachLength) {
-                        currIter.pop();
-                        const rebasedMark = clone(currMark);
-
-                        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-                        removeLineageEvent(rebasedMark, baseRevision!);
-                        factory.pushContent(rebasedMark);
-                        lineageRequests.push({ mark: rebasedMark, offset });
-                    }
-                } else if (isAttachAfterBaseAttach(currMark, baseMark)) {
-                    baseIter.pop();
-                    factory.pushOffset(getOutputLength(baseMark));
-                } else {
-                    currIter.pop();
-                    handleCurrAttach(currMark, factory, lineageRequests, baseDetachOffset);
-                }
-            } else {
-                currIter.pop();
-                handleCurrAttach(currMark, factory, lineageRequests, baseDetachOffset);
-            }
-        } else if (isAttach(baseMark)) {
-            baseIter.pop();
-            factory.pushOffset(getOutputLength(baseMark));
         } else {
-            // If we've reached this branch then `baseMark` and `currMark` start at the same location
-            // in the document field at the revision to which both changesets apply.
-            // Despite that, it's not necessarily true that they affect the same range in that document
-            // field because they may be of different lengths.
-            // We perform any necessary splitting in order to end up with a pair of marks that do have the same length.
-            currIter.pop();
-            baseIter.pop();
-            baseReattachOffset = 0;
-            const currMarkLength = getInputLength(currMark);
-            const baseMarkLength = getInputLength(baseMark);
-            if (currMarkLength < baseMarkLength) {
-                let nextBaseMark;
-                [baseMark, nextBaseMark] = splitMarkOnInput(baseMark, currMarkLength);
-                baseIter.push(nextBaseMark);
-            } else if (currMarkLength > baseMarkLength) {
-                let nextCurrMark;
-                [currMark, nextCurrMark] = splitMarkOnInput(currMark, baseMarkLength);
-                currIter.push(nextCurrMark);
-            }
-            // Past this point, we are guaranteed that `baseMark` and `currMark` have the same length and
-            // start at the same location at the revision to which both changesets apply.
-            // They therefore refer to the same range for that revision.
+            assert(
+                !isAttach(baseMark) && !isAttach(currMark),
+                "Two attaches cannot be at the same position",
```

Comment:
The condition is different from what the error message implies.
The condition asserts that neither mark is an attach.
The message implies it's okay for one to be an attach so long as the second isn't.

Reply:
I think we indeed don't expect either of them to be an insert.
In the future I think it'll be possible for either or both of them to be reattaches.